x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/perform_rule_upgrade/perform_rule_upgrade_route.schema.yaml

```diff
@@ -0,0 +1,507 @@
+openapi: 3.0.0
+info:
+  title: Perform Rule Upgrade API endpoint
+  version: '2023-10-31'
+paths:
+  /api/detection_engine/rules/prebuilt/_perform_upgrade:
+    post:
+      x-labels: [ess]
+      x-codegen-enabled: true
+      operationId: PerformRuleUpgrade
+      summary: Perform rule upgrade
+      description: Upgrade prebuilt detection rules.
+      tags:
+        - Rules API
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: '#/components/schemas/UpgradeAllRulesRequest'
+                - $ref: '#/components/schemas/UpgradeSpecificRulesRequest'
+      responses:
+        200:
+          description: Indicates a successful call.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PerformRuleUpgradeResponseBody'
+
+components:
+  schemas:
+    RulePickVersionValues:
+      type: string
+      enum: [BASE, CURRENT, TARGET, MERGED]
+
+    FieldPickVersionValues:
+      type: string
+      enum: [BASE, CURRENT, TARGET, MERGED, RESOLVED]
+
+    RuleUpgradeSpecifier:
+      type: object
+      required:
+        - rule_id
+        - revision
+        - version
+      properties:
+        rule_id:
+          $ref: '../../model/rule_schema/common_attributes.schema.yaml#/components/schemas/RuleSignatureId'
+        revision:
+          type: number
+        version:
+          $ref: '../../model/rule_schema/common_attributes.schema.yaml#/components/schemas/RuleVersion'
+        pick_version:
+          $ref: '#/components/schemas/RulePickVersionValues'
+        fields:
+          type: object
+          description: |
+            Fields that can be customized during the upgrade workflow
+            as decided in: https://github.com/elastic/kibana/issues/186544
+            Fields listed here, which are not specified in the request body,
+            will default to a `pick_version` of `MERGED`.
+          properties:
+            name:
+              type: object
+              required:
+                - pick_version
+              properties:
+                pick_version:
```

Comment:
Should it be `oneOf`? Either `pick_version` or `resolved_value` need to be specified?

Reply:
No, `resolved_value` should only be defined if `pick_version` is `RESOLVED`.